### ISSUE TITLE
Element Instead of Window Responsiveness

### ISSUE THREAD

**Issue by imjoshdean-tc** [akiran/react-slick#1159]
Hey there, I am running into a situation where I need to create responsiveness based off of a parent element for my slider rather than the window. Is there a way to do this in your current implementation? Doing some digging I saw this I saw https://github.com/akiran/react-slick/pull/1107 was created and closed that helps solve this issue.

Is there any hope for getting this into the repo anytime soon, and if not is there a way to handle it with what is currently out in 0.22?

**Comment by laveesingh** [comment]
That resize observer has already been implemented. And it is already there in version 0.22.

**Comment by imjoshdean-tc** [comment]
I saw that the resize observer is already there which is great! I wanted to know if there was a timeline for responsiveness on elements rather than window, or if you have a suggestion for an implementation of that, that would be fantastic as well!

**Comment by laveesingh** [comment]
I'm not sure if I understand what you mean by timeline.

**Comment by imjoshdean-tc** [comment]
Sorry, let me rephrase. I wanted to know if you had any idea of when element responsiveness would be added as a feature. I see that you added an "On Deck" label, so I'm guessing it's something you'll be working on soon?

I'm working on utilizing react-slick for a department store carousel that has other elements that cause the carousel to shrink and grow in size, and being able to have responsive breakpoints on the element wrapping the carousel would go a long way to helping me get it done quicker.

**Comment by laveesingh** [comment]
You should try the version 0.22, this element responsiveness feature is already there. Anyway, the next release is due today.

**Comment by sag1v** [comment]
I'm facing the same issue with the latest version. 
Seems that the breakpoint is chosen by the window's `with` and not the element itself.

**Comment by sag1v** [comment]
#1212 should address this issue
